docs/javascript.md

```diff
@@ -124,6 +124,166 @@ if (isLoggedIn) {
+    this.baseUrl = baseUrl;
+    this.fetchingService = axios;
+    this.apiVersion = apiVersion
+  }
+
+  private getFullApiUrl(url) {
+    return `${this.baseUrl}/${this.apiVersion}/${url}`;
+  }
+
+  private populateTokenToHeaderConfig() {
+    return {
+      'Authorization': localStorage.getItem('token'),
+    }
+  }
+  
+  private extractUrlAndDataFromConfig({data, url, ...configWithoutDataAndUrl}) {
+    return configWithoutDataAndUrl;
+  }
+
+  get(config, withAuth = true) {
+    if (withAuth) {
+      config.headers = {
+        ...config.headers,
+        ...populateTokenToHeaderConfig(),
+      }
+    }
+    return this.fetchingService.get(this.getFullApiUrl(config.url), extractUrlAndDataFromConfig(config));
+  }
+
+  put(config, withAuth = true) {
+    if (withAuth) {
+      config.headers = {
+        ...config.headers,
+        ...populateTokenToHeaderConfig(),
+      }
+    }
+    return this.fetchingService.get(this.getFullApiUrl(config.url), config.data, extractUrlAndDataFromConfig(config));
+  }
+}
+```
+```javascript
+// user.service.js
+import HttpService from './http.service';
+
+class UserService extends HttpSerivce {
+  constructor() {
+    super();
+  }
+  getUsers() {
+    return this.get({
+      url: 'users',
+    })
+  }
+  updateUser(user) {
+    return this.post({
+      url: 'users',
+      data: user,
+  })
+}
+```
+- ### Always try create models to serialize/deserialize your etities
+> Instead of
+
+```javascript
+// user.service.js
+class UserService extends HttpSerivce {
+  constructor() {
+    super();
+  }
+  async getUsers() {
+    const response = await this.get({
+      url: 'users',
+    });
+    return response.data.map(user => ({
+      ...user,
+      fullName: user.full_name,
+      dayOfBirth: new Date(user.dob),
+    }))
+  }
+}
+```
+
+> Better to use
+
+```javascript
+// user.model.js
+
+class UserModel {
+  constructor(id, fullName, dayOfBirth, status) {
+    this.id = user.id;
+    this.fullName = user.full_name;
+    this.dayOfBirth =  new Date(user.dob);
+    this.status = user.status;
+  }
+  
+  getFormatedDayOfBirth() {
+    return this.dayOfBirth.toDateString();
+  }
+}
+
+const createUserModel = (userFromServer) => {
+  return new UserModel(userFromServer.id, userFromServer.full_name, userFromServer.dob, userFromServer.status);
+};
+
+export {createUserModel};
+
+export default UserModel;
+
+```
+```javascript
+  // user.service.js
+  import { createUserModel } from './user.model';
+
+  class UserService extends HttpSerivce {
+    constructor() {
+      super();
+    }
+    async getUsers() {
+      const response = await this.get({
+        url: 'users',
+      });
+      return response.data.map(user => createUserModel());
+    }
+}
+```
+> With that aproach you works with all data that comes from server like a Etities. So your code will have better abstraction level. Also as you see you can atach usefull methods to a etity model.

```

Comment:
With that approach you can work with all the data that comes from your server, such as entities. So your code will have better level of abstraction. Also as you see you can attach useful methods to the entity model.